src/lndUtils.ts

```diff
@@ -444,52 +439,38 @@ export const validate = async ({
         expires_at,
         features,
       } = await parsePaymentRequest({ request: params.invoice }))
-      // ({ id, safe_tokens: tokens, destination, description, routes: routeHint, payment, cltv_delta, expires_at, features } = await decodePaymentRequest({ lnd, request: params.invoice }))
     } catch (err) {
       const error = `Error decoding the invoice`
       logger.error({ params, success: false, error }, error)
       throw new LoggedError(error)
     }
 
     // TODO: if expired_at expired, thrown an error
-
-    if (!!params.amount && !!tokens) {
-      const error = `Invoice contains non-zero amount, but amount was also passed separately`
-      // FIXME: create a new error. this is a not a graphl error.
-      // throw new ValidationError(error, {logger})
-
-      throw new ValidationError(error)
-    }
   } else {
     if (!params.username) {
-      // FIXME: create a new error. this is a not a graphl error.
-      // throw new ValidationError(error, {logger})
-
       const error = `a username is required for push payment to the ${yamlConfig.name}`
-      throw new ValidationError(error)
+      throw new ValidationInternalError(error, { logger })
     }
 
     isPushPayment = true
-    destination = params.destination
     username = params.username
+  }
 
-    const preimage = randomBytes(preimageByteLength)
-    id = createHash("sha256").update(preimage).digest().toString("hex")
-    const secret = preimage.toString("hex")
-    messages = [{ type: keySendPreimageType, value: secret }]
+  if (!!params.amount && !!tokens) {
+    const error = `Invoice contains non-zero amount, but amount was also passed separately`
+    // FIXME: create a new error. this is a not a graphl error.
+    // throw new ValidationInternalError(error, {logger})
```

Comment:
This comment is duplicate of the line below??

Reply:
no the condition are different. it's basically an xor we are doing here, which an explicit error on whether we have too much info or too little